### ISSUE TITLE
Provide an AppImage for Linux

### ISSUE THREAD

**Issue by probonopd** [trufont/trufont#624]
Providing an [AppImage](http://appimage.org/) would have, among others, these advantages:
- Applications packaged as an AppImage can run on many distributions (including Ubuntu, Fedora, openSUSE, CentOS, elementaryOS, Linux Mint, and others)
- One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
- No unpacking or installation necessary
- No root needed
- No system libraries changed
- Works out of the box, no installation of runtimes needed
- Optional desktop integration with `appimaged`
- Optional binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
- Can optionally GPG2-sign your AppImages (inside the file)
- Works on Live ISOs
- Can use the same AppImages when dual-booting multiple distributions
- Can be listed in the [AppImageHub](https://appimage.github.io/apps) central directory of available AppImages
- Can double as a self-extracting compressed archive with the `--appimage-extract` parameter
- No repositories needed. Suitable/optimized for air-gapped (offline) machines

[Here is an overview](https://appimage.github.io/apps) of projects that are already distributing upstream-provided, official AppImages.

This script is the beginning of what might become an automated way to generate a fresh AppImage for each `git push`:

```
#! /bin/bash

sudo apt -y install git

export APPNAME=trufont
export PIP_REQUIREMENTS="-e git+https://github.com/trufont/trufont#egg=$APPNAME fs"
# export CONDA_PACKAGES="$APPNAME" # Only use this if the app is in a Conda channel (e.g., conda-forge)

wget -c "https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-x86_64.AppImage"
wget -c "https://raw.githubusercontent.com/TheAssassin/linuxdeploy-plugin-conda/master/linuxdeploy-plugin-conda.sh"
chmod +x linuxdeploy-x86_64.AppImage linuxdeploy-plugin-conda.sh

rm -r AppDir || true

cat > $APPNAME.desktop <<EOF
[Desktop Entry]
Version=1.0
Name=TruFont
Exec=trufont %U
Terminal=false
Type=Application
Icon=$APPNAME
Categories=Graphics;
EOF

cat > AppRun <<\EOF
#!/bin/bash
HERE="$(dirname "$(readlink -f "${0}")")"
exec "$HERE/usr/conda/bin/python" "$HERE/usr/conda/bin/trufont" "$@"
EOF
chmod +x AppRun

wget -c "https://github.com/trufont/trufont/raw/master/Lib/trufont/resources/app.png"
convert app.png -resize 512x512 $APPNAME.png

./linuxdeploy-x86_64.AppImage --appdir AppDir --plugin conda -i $APPNAME.png -d $(readlink -f "$APPNAME.desktop") --custom-apprun AppRun --output appimage
```
![](https://user-images.githubusercontent.com/2480569/50595380-f1bc2280-0e97-11e9-80d3-d6c4f5c3f3ff.png)

The resulting AppImage is not optimized for size yet, but seems to run well on Xubuntu 18.04:

![](https://user-images.githubusercontent.com/2480569/50595362-de10bc00-0e97-11e9-8c98-ccd1bc285f17.png)

If you have questions, AppImage developers are on #AppImage on irc.freenode.net.

**Comment by adrientetar** [comment]
@probonopd yes thanks, just be aware new developments are happening on the wx branch, which is catching up in features with master.

**Comment by probonopd** [comment]
So @adrientetar should I send a PR for the wx branch instead? Would you consider to merge it then? Thanks.

**Comment by probonopd** [comment]
Hi @adrientetar, looking at the activity in this repository it seems like the master branch is seeing more development than the wx branch lately. So should I send a PR for the master branch nowadays?

**Comment by madig** [comment]
The WX effort is currently abandoned.
